pymatgen/analysis/quasirrho.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License.
+
+
+"""
+A module to calculate free energies using the Quasi-Rigid Rotor Harmonic
+ Oscillator approximation. Modified from a script by Steven Wheeler.
+See: Grimme, S. Chem. Eur. J. 2012, 18, 9955
+"""
+
+from __future__ import annotations
+
+__author__ = "Alex Epstein"
+__copyright__ = "Copyright 2020, The Materials Project"
+__version__ = "0.1"
+__maintainer__ = "Alex Epstein"
+__email__ = "aepstein@lbl.gov"
+__date__ = "August 1, 2023"
+__credits__ = "Ryan Kingsbury, Steven Wheeler, Trevor Seguin, Evan Spotte-Smith"
+
+from math import isclose
+from typing import TYPE_CHECKING
+
+import numpy as np
+import scipy.constants as const
+
+from pymatgen.core.units import kb as kb_ev
+
+if TYPE_CHECKING:
+    from pymatgen.core import Molecule
+    from pymatgen.io.gaussian import GaussianOutput
+    from pymatgen.io.qchem.outputs import QCOutput
+
+# Define useful constants
+kb = kb_ev * const.eV  # Pymatgen kb [J/k]
+c = const.speed_of_light * 100  # [cm/s]
+h = const.h  # Planck's constant [J.s]
+R = const.R / const.calorie  # Ideal gas constant [cal/mol/K]
+R_ha = const.R / const.value("Hartree energy") / const.Avogadro  # Ideal gas
+# constant [Ha/K]
+R_volume = const.R / const.atm * 1000  # Ideal gas constant [L.atm.K^-1.mol^-1]
+
+# Define useful conversion factors
+amu_to_kg = const.value("atomic mass unit-kilogram relationship")  # AMU to kg
+# kcal2hartree = 0.0015936  # kcal/mol to hartree/mol
+kcal2hartree = 1000 * const.calorie / const.value("Hartree energy") / const.Avogadro
+
+
+def get_avg_mom_inertia(mol):
```

Comment:
Not required for this PR, but I wonder if this should be made a @property of the `Molecule` class? Any strong opinions one way or another?